runtime/runtime/src/lib.rs

```diff
@@ -846,13 +913,19 @@ impl Runtime {
     /// receivers) and incoming action receipts.
     pub fn apply(
         &self,
-        mut state_update: TrieUpdate,
+        trie: Arc<Trie>,
+        root: CryptoHash,
+        validator_accounts_update: &Option<ValidatorAccountsUpdate>,
         apply_state: &ApplyState,
         prev_receipts: &[Receipt],
         transactions: &[SignedTransaction],
     ) -> Result<ApplyResult, InvalidTxErrorOrStorageError> {
-        // TODO(#1481): Remove clone after Runtime bug is fixed.
-        let initial_state = state_update.clone();
+        let initial_state = TrieUpdate::new(trie.clone(), root);
```

Comment:
@Kouprin Is working on trie parts. His abstraction requires encapsulating root in `StateRoot` instead of exposing it as a hash directly. We might want to avoid working with root hashes directly in the runtime.

Reply:
I was wondered what exactly should be used here. Seems most places still use hash, so I decided to use the hash instead of the `StateRoot`.